warcraft/contracts/assets/stateful.py

```diff
@@ -0,0 +1,62 @@
+from pyteal import *
+
+def approval_program():
+    app_manager = Bytes("app_manager")
+    total_registered = Bytes("total_registered")
+    escrow = Bytes("escrow")
+    # verify rekey_to and close_rem_to are set as zero_address
+    basic_checks = And(
+        # Always verify that the RekeyTo property of any transaction is set to the ZeroAddress
+        # unless the contract is specifically involved in a rekeying operation.
+        Txn.rekey_to() == Global.zero_address(),
+        Txn.close_remainder_to() == Global.zero_address(),
+        Txn.asset_close_to() == Global.zero_address()
+    )
+
+    on_initialize = Seq([
+        App.globalPut(app_manager, Txn.application_args[0]),
```

Comment:
Why do we store the `app_manager` in state, if we never update it?

Reply:
because we can retrieve it while in a script, although in terms of optimization(not much optimized though) it could be removed but manager should be publicly available using one call, instead of scrapping public code and try to find the manager

Reply:
My question is do we ever need to update `app_manager` here?

+ If no, then this should be a template parameter. It's about saving cost (each global byte has some cost associated with it). I think generally for fixed values, hardcoding is better. What is the specific use case that we would retrieve `app_manager` in a script ? BDW, maybe in this case `app_manager` should be deployer(creator) of app, because if we never update it, then while deploying one user can deploy this app while setting another address as manager. And it could be easily retrieved by `Global.creatorAddress`. 

+ If yes, we should have a branch `update_manager`: existing manager can change app_manager to a new address.

Reply:
And my answer is about a general Dapp present on network and not a Dapp in a template. (While testing you can get any address you want but things change when it's on mainnet) My answer above is for general, but it doesn't make sense to discuss it on this template.

Reply:
Let's hardcode it.